### PR TITLE
Drop NetworkId type from domain

### PR DESCRIPTION
**Pull request by MartinquaXD**
## Changes
`NetworkId` and `ChainId` are the same thing so it only makes sense to keep one.
* drop `NetworkId`
* drop `Network` since it would only have 1 field
* switch internal value of `ChainId` from `U256` to `u64` since this is guaranteed to fit per [EIP-2294](https://github.com/ethereum/EIPs/issues/2294)
* since `ChainId` is now same size as a pointer I made all functions take an owned value instead of a reference

## How to test
tests, compiler

## Related to https://github.com/cowprotocol/services/issues/2414

### CODE REVIEW

File: crates/autopilot/src/infra/blockchain/mod.rs [hunk 1]
```diff
@@ -16,57 +16,37 @@ pub mod contracts;
...
     web3: DynWeb3,
-    network: Network,
-}
-
-/// Network information for an Ethereum blockchain connection.
-#[derive(Clone, Debug)]
-pub struct Network {
-    pub id: NetworkId,
-    pub chain: ChainId,
+    network: ChainId,
```
Comment (fleupold) [inline crates/autopilot/src/infra/blockchain/mod.rs:34]:
```suggestion
    chain: ChainId,
```

**Review by fleupold — APPROVED** [review]
Can we unify the naming as well and call everything either chain or network (slight preference for chain)?

**Review by sunce86 — APPROVED** [review]
Not related to this PR, but implementing Deref for most of the structs like `StructName(pub U256)` might be nicer to use at the callsite. If anyone shares the opinion we should probably do it for all of them.

**Comment by MartinquaXD** [comment]
FYI I merged the PR stack in the wrong order by accident so this diff now also contains changes from the other PR. 🤦‍♂️

**Pull request merged**